docs/sdk/src/reference_docs/frame_currency.rs

```diff
@@ -1,8 +1,55 @@
-//! FRAME Currency Abstractions and Traits
+//! # FRAME Currency Abstractions and Traits
 //!
-//! Notes:
+//! Currency related traits in FRAME provide standardized interfaces for implementing various
+//! currency functionalities. The fundamental reason to have these traits is to allow pallets to
+//! utilize a token without relying on the implementation detail. Or else we could have a perfectly
+//! fine runtime without these traits.
+//! These traits enable developers to create, transfer, and manage different forms of digital assets
+//! within the blockchain environment, ensuring that the economic aspects of the chain are robust
+//! and adaptable to different use cases.
+
 //!
-//! - History, `Currency` trait.
-//! - `Hold` and `Freeze` with diagram.
-//! - `HoldReason` and `FreezeReason`
-//! 	- This footgun: <https://github.com/paritytech/polkadot-sdk/pull/1900#discussion_r1363783609>
+//! ## The Currency Trait
+//!
+//! The [`Currency`][4] trait was initially introduced in Substrate to manage the native token
+//! balances. This trait was later deprecated in favor of the [`fungible`][3] traits in Substrate's
+//! PR [#12951](https://github.com/paritytech/substrate/pull/12951), the main reason for this is
+//! that the deprecated trait is only sensible in a system that has just one currency type. This
+//! shift is part of a broader initiative to enhance token management capabilities within the
+//! framework. This deprecation is aimed at providing improved safety and more flexibility for
+//! managing assets, beyond the capabilities of the original [`Currency`][4] trait. This transition
+//! enables more diverse economic models in Substrate. For more details, you can view the discussion
+//! on the [Tokens Horizon issue](https://github.com/paritytech/polkadot-sdk/issues/327). The
+//! [`Currency`][4] trait is still available in Substrate, but it is recommended to use the
+//! **fungible** traits instead. The [deprecation PR](https://github.com/paritytech/substrate/pull/12951)
+//! has a dedicated section on upgrading from **Currency** to **fungible**. Besides, this
+//! [issue](https://github.com/paritytech/polkadot-sdk/issues/226) lists the pallets that have been
+//! upgraded to the **fungible** traits, and the ones that are still using the [`Currency`][4]
+//! trait. There one could find the relevant code examples for upgrading.
+//!
+//! ## Fungible Traits
+//!
+//! The [`fungible`][3] traits are designed for managing currency types, providing a streamlined
+//! approach for single-asset operations. Fungible is currently preferred over currency as the
+//! latter is deprecated.
+//!
+//! #### Holds and Freezes
+//!
+//! Learn more about this two concepts in
+//! [`frame_support::traits::tokens::fungible::hold`][1] and
+//! [`frame_support::traits::tokens::fungible::freeze`][2].
```

Comment:
```suggestion
//! [`frame_support::traits::tokens::fungible::freeze`].
```

I believe the link is added automatically in this case?

Reply:
Not really as `frame_support` is not a dependency of `polkadot-sdk-docs`

Reply:
Should be able to add it as a dep?